elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/Column.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.metadata.models;
+
+import com.yahoo.elide.datastores.aggregation.AggregationDictionary;
+import com.yahoo.elide.datastores.aggregation.metadata.enums.Tag;
+import com.yahoo.elide.datastores.aggregation.metadata.enums.ValueType;
+
+import lombok.Data;
+import lombok.ToString;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+
+/**
+ * Column is the super class of a field in a table, it can be either dimension or metric.
+ */
+@Data
+@ToString
+public abstract class Column {
+    @Id
+    private String id;
+
+    private String name;
+
+    private String longName;
+
+    private String description;
+
+    private String category;
+
+    @ManyToMany
+    private DataType dataType;
```

Comment:
We can construct a data type on our own, using annotations. But I think this needs to be a relationship so that we can track all columns with the same type.